### DISCUSSION TITLE
Tests are collected in the root, but not in a subfolder

### DISCUSSION THREAD

**Discussion by IlyaFaer** [pytest-dev/pytest#9994] [Q&A]
Hi, all!

Met an uneasy error. There is a Python package with a `test` subfolder. In the subfolder there is a file `test_suite_13.py` with a dummy test:
```python
class TestAbc:
    def test_abd(self):
        assert 1 == 1

```
When I'm trying to run the test with a simple command `pytest test/test_suite_13.py` I'm getting `collected 0 items`. The same applies to the command `pytest test`.

But if I'll move this file to the root of the package and run `pytest test_suite_13.py` it works just fine!

Of course, I know the pytest naming conventions, I've tried different variants of a configuration file and I've also tried to rename files and folders to something else (starting and ending with `test`, lack of `test`, etc.). Nothing is working. What can you advice?

P.S.
The most weird thing is that in fact the test file is way more complex than the dummy script I've posted, and there are two test files (`test_suite_13.py` and `test_suite_14.py`) in my `test` folder, almost they same, just used for testing different versions of the same package. They have insignificant difference deep in the base classes, but they are executed in the same way. And still, one of them is collected correctly and all the tests are passing, while another one is always `collected 0 items`. No meaningful difference between them noticed, so we're quite lost here.

Python version: 3.7
Pytest version: the last one
OS: both Windows 11 and the last Linux

**Comment by The-Compiler** [comment]
Can you show the full output of when it works and when it does not?

**Comment by IlyaFaer** [comment]
This one is when the test file is located in the root directory of the package:

![Screenshot 2022-05-26 131224](https://user-images.githubusercontent.com/26030071/170457405-68b81fbd-b10c-4729-8174-3f9efb8b5bea.png)

This one is when the test file is in the subfolder:

![2](https://user-images.githubusercontent.com/26030071/170457549-417d3526-6455-448b-b499-55ce9dc3691b.png)

**Reply by The-Compiler** [reply]
What's in the `test/conftest.py`? Any pytest hooks?

**Reply by IlyaFaer** [reply]
Here is the full `conftest.py` code:

```python
import configparser
import datetime
import os
import uuid

import pytest

from sqlalchemy import (
    Column,
    Integer,
    MetaData,
    String,
    Table,
    create_engine,
    ForeignKey,
)


@pytest.fixture
def db_url():
    project = os.getenv(
        "GOOGLE_CLOUD_PROJECT", os.getenv("PROJECT_ID", "emulator-test-project"),
    )
    db_url = (
        f"spanner:///projects/{project}/instances/"
        "sqlalchemy-dialect-test/databases/compliance-test"
    )

    config = configparser.ConfigParser()
    if os.path.exists("test.cfg"):
        config.read("test.cfg")
    else:
        config.read("setup.cfg")
    return config.get("db", "default", fallback=db_url)


@pytest.fixture
def table_id():
    now = datetime.datetime.now()
    table_id = "example_table_{}_{}".format(
        now.strftime("%Y%m%d%H%M%S"), uuid.uuid4().hex[:8]
    )
    return table_id


@pytest.fixture
def table(db_url, table_id):
    engine = create_engine(db_url)
    metadata = MetaData(bind=engine)

    table = Table(
        table_id,
        metadata,
        Column("user_id", Integer, primary_key=True),
        Column("user_name", String(16), nullable=False),
    )
    table.create()
    yield table
    table.drop()


@pytest.fixture
def table_w_foreign_key(db_url, table):
    engine = create_engine(db_url)
    metadata = MetaData(bind=engine)

    table_fk = Table(
        "table_fk",
        metadata,
        Column("id", Integer, primary_key=True),
        Column("name", String(16), nullable=False),
        Column(
            table.name + "_user_id",
            Integer,
            ForeignKey(table.c.user_id, name=table.name + "user_id"),
        ),
    )
    table_fk.create()
    yield table_fk
    table_fk.drop()


@pytest.fixture
def connection(db_url):
    engine = create_engine(db_url)
    return engine.connect()


def insert_data(conn, table, data):
    conn.execute(table.insert(), data)
```

**Reply by IlyaFaer** [reply]
And here is the `setup.cfg`:

```python
[tool:pytest]
addopts= --tb native -v -r fxX --maxfail=25 -p no:warnings
python_files=test_*.py

[sqla_testing]
requirement_cls=google.cloud.sqlalchemy_spanner.requirements:Requirements
profile_file=test/profiles.txt
```
